src/controllers/confirmation.controller.ts

```diff
@@ -51,25 +51,31 @@ const route = async (req: Request, res: Response, next: NextFunction): Promise<v
       });
   }
 
-  const token = req.chSession.accessToken() as string;
-  let apiResponseData: any;
-  try {
-    // TODO  LFA-1406 Add isSubmitted flag to prevent this being sent twice
+  const isSubmitted: boolean = getPromiseToFileSessionValue(req.chSession, ALREADY_SUBMITTED);
+  if (!isSubmitted) {
+    const token = req.chSession.accessToken() as string;
+    let apiResponseData: any;
+    try {
 
-    const axiosResponse: AxiosResponse = await callPromiseToFileAPI(companyProfile.companyNumber,
-        token, isStillRequired);
+        const axiosResponse: AxiosResponse = await callPromiseToFileAPI(companyProfile.companyNumber,
+            token, isStillRequired);
 
-    apiResponseData = axiosResponse.data;
+        apiResponseData = axiosResponse.data;
+        await updatePromiseToFileSessionValue(req.chSession, NEW_DEADLINE, apiResponseData.filing_due_on);
```

Comment:
I don't understand why this is being done. It isn't mentioned in the requirements/story and why does it need to be stored in the session?

Reply:
As stated above even when submitted is set to true the code still needs the new deadline so that it can be displayed on the confirmation screen the only thing that should be different is the api client is not called. Therefore we need somewhere global to access the new deadline data.